tap/tapify.py

```diff
@@ -253,44 +257,58 @@ def _tap_data(class_or_function: _ClassOrFunction, param_to_description: Dict[st
     return _tap_data_from_class_or_function(class_or_function, func_kwargs, param_to_description)
 
 
-def _tap_class(args_data: Sequence[_ArgData]) -> Type[Tap]:
+def _tap_class(args_data: Sequence[_ArgData], all_args_named: bool = True) -> Type[Tap]:
+    """
+    Transfers argument data to a :class:`tap.Tap` class.
+    """
+
     class ArgParser(Tap):
         # Overwriting configure would force a user to remember to call super().configure if they want to overwrite it
         # Instead, overwrite _configure
         def _configure(self):
             for arg_data in args_data:
                 variable = arg_data.name
+                # Set internal Tap attributes
                 self._annotations[variable] = str if arg_data.annotation is Any else arg_data.annotation
                 self.class_variables[variable] = {"comment": arg_data.description or ""}
-                if arg_data.is_required:
-                    kwargs = {}
+
+                # Add the argument to the parser
+                if not all_args_named and (arg_data.is_positional_only or arg_data.is_required):
```

Comment:
Included the `or arg_data.is_required` b/c it might be nice to have the option to have required arguments show up as positional-only in the command line so that all required arguments show up at the top of the help message. Lmk if you prefer to remove this

Reply:
I agree that it is good to have the option to make required arguments show up as positional only. It  would make reading the help message and writing command from the terminal more convenient.